src/name2dim.jl

```diff
@@ -0,0 +1,47 @@
+
+function NamedNotFoundException(name, cands)
+    return ArgumentError(
+        "No dimensioned called $name exists. Dimension names: $(join(cands, ", "))"
+    )
+end
+
+###################################
+
+# We change the name into a `Val` because that will trigger constant propergation
```

Comment:
This doesn't need to be a val for constant propagation, just for dispatch

Reply:
Correct comment was incorrect notes from earlier experiments in making this compile time